src/main.rs

```diff
@@ -38,21 +39,25 @@ fn main() -> ExitCode {
     let errors: HashMap<String, Option<Diagnostic>> = args
         .files
         .into_iter()
-        .map(|f: String| -> (String, Option<Diagnostic>) {
-            let key = &f;
-            (
-                key.to_string(),
-                fs::File::open(&f)
-                    .map(|ref mut file| analyse_sql(file))
-                    .expect("failed to open file"),
+        .flat_map(|f| {
+            WalkDir::new(f).into_iter().filter_map(|e| e.ok()).map(
```

Comment:
Maybe we should filter items whose extension is not `.sql`, IMO.
Currently, `analyse_sql()` pass through non-sql files.

I think following feature is useful for it.
https://docs.rs/walkdir/latest/walkdir/struct.FilterEntry.html

What do you think?

Reply:
I'm sorry for the late reply. I fixed it!